service/worker/scanner/scanner.go

```diff
@@ -137,20 +143,25 @@ func (s *Scanner) Start() error {
 
 	var workerTaskQueueNames []string
 	if s.context.cfg.ExecutionsScannerEnabled() {
+		s.wg.Add(1)
 		go s.startWorkflowWithRetry(executionsScannerWFStartOptions, executionsScannerWFTypeName)
 		workerTaskQueueNames = append(workerTaskQueueNames, executionsScannerTaskQueueName)
 	}
 
 	if s.context.cfg.Persistence.DefaultStoreType() == config.StoreTypeSQL && s.context.cfg.TaskQueueScannerEnabled() {
+		s.wg.Add(1)
 		go s.startWorkflowWithRetry(tlScannerWFStartOptions, tqScannerWFTypeName)
 		workerTaskQueueNames = append(workerTaskQueueNames, tqScannerTaskQueueName)
-	} else if s.context.cfg.Persistence.DefaultStoreType() == config.StoreTypeNoSQL && s.context.cfg.HistoryScannerEnabled() {
+	}
+
+	if s.context.cfg.HistoryScannerEnabled() {
+		s.wg.Add(1)
 		go s.startWorkflowWithRetry(historyScannerWFStartOptions, historyScannerWFTypeName)
 		workerTaskQueueNames = append(workerTaskQueueNames, historyScannerTaskQueueName)
 	}
 
 	for _, tl := range workerTaskQueueNames {
-		work := worker.New(s.context.sdkSystemClient, tl, workerOpts)
+		work := s.context.workerFactory.New(s.context.sdkSystemClient, tl, workerOpts)
```

Comment:
what is the difference here?

Reply:
Injecting a worker factory here allows me to test which workers were created. If I just use `worker.New`, it does some assertion that the worker is created by a real sdk client.